### PR TITLE
Bump dep

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

I am targeting this branch, because {reason}.

Closes #{put_issue_number_here}.

## Changelog

```markdown
### Removed
- Support for doctrine/persistence 1
- Support for php 7.4
```

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -21,9 +21,9 @@
     ],
     "homepage": "https://github.com/sonata-project/sonata-doctrine-extensions",
     "require": {
-        "php": "^7.4 || ^8.0",
+        "php": "^8.0",
         "doctrine/dbal": "^2.13 || ^3.1",
```
Comment (jordisala1991) [inline composer.json]: we could also drop dbal 2

File: src/Mapper/Builder/OptionsBuilder.php [hunk 2]
```diff
@@ -25,15 +25,11 @@ final class OptionsBuilder
      */
     private array $options = [];
 
-    private ?string $type;
-
     /**
      * NEXT_MAJOR: Make the arguments mandatory.
      */
-    private function __construct(?string $type = null, ?string $fieldName = null, ?string $targetEntity = null)
+    private function __construct(private ?string $type = null, ?string $fieldName = null, ?string $targetEntity = null)
```
Comment (jordisala1991) [inline src/Mapper/Builder/OptionsBuilder.php]: multiline

**Review by jordisala1991 — DISMISSED** [review]
Build is failing tho

File: tests/Mapper/ORM/DoctrineORMMapperTest.php [hunk 3]
```diff
@@ -52,13 +52,7 @@ public function testDoctrineMappingLoaded(): void
...
         self::bootKernel();
 
-        /**
-         * TODO: Simplify when Symfony 4.4 support is dropped.
-         *
-         * @phpstan-ignore-next-line
-         * @psalm-suppress UndefinedPropertyFetch
-         */
-        $container = method_exists($this, 'getContainer') ? static::getContainer() : static::$container;
+        $container = static::getContainer();
```
Comment (jordisala1991) [inline tests/Mapper/ORM/DoctrineORMMapperTest.php]: I guess you can avoid OTV (one time variable)

**Comment by VincentLanglet** [comment]
> Build is failing tho

Any idea why @jordisala1991 ?
I tried bumping multiples dependencies without success

File: composer.json [hunk 4]
```diff
@@ -21,17 +21,17 @@
...
     "require": {
-        "php": "^7.4 || ^8.0",
-        "doctrine/dbal": "^2.13 || ^3.1",
-        "doctrine/persistence": "^1.3.6 || ^2.0 || ^3.0"
+        "php": "^8.0",
+        "doctrine/dbal": "^3.3.6",
+        "doctrine/persistence": "^2.0 || ^3.0"
     },
     "require-dev": {
         "doctrine/common": "^2.7 || ^3.0",
```
Comment (jordisala1991) [inline composer.json:29]: this should be bumped too right?

**Comment by jordisala1991** [comment]
> > Build is failing tho
> 
> Any idea why @jordisala1991 ? I tried bumping multiples dependencies without success

Maybe the minimum of2.0 of persistence?

**Comment by VincentLanglet** [comment]
> > > Build is failing tho
> > 
> > 
> > Any idea why @jordisala1991 ? I tried bumping multiples dependencies without success
> 
> Maybe the minimum of2.0 of persistence?

I found a way to solve them.

**Review by jordisala1991 — APPROVED** [review]

File: composer.json [hunk 5]
```diff
@@ -21,41 +21,41 @@
...
-        "doctrine/doctrine-bundle": "^2.0",
-        "doctrine/mongodb-odm": "^2.0",
-        "doctrine/orm": "^2.5",
+        "doctrine/doctrine-bundle": "^2.6",
+        "doctrine/mongodb-odm": "^2.2",
+        "doctrine/orm": "^2.10",
         "friendsofphp/php-cs-fixer": "^3.4",
-        "jackalope/jackalope-doctrine-dbal": "^1.0",
+        "jackalope/jackalope-doctrine-dbal": "^1.7.6",
```
Comment (franmomu) [inline composer.json:34]: is this dependency still needed? I think it was used along with phpcr

Comment (VincentLanglet) [reply composer.json:34]: On it https://github.com/sonata-project/sonata-doctrine-extensions/pull/417

**Pull request merged**